### PR TITLE
bv2int: refactoring the main translation loop

### PR DESCRIPTION
**Pull request by yoni206**
This PR introduces a refactoring of the main translation loop in bv2int preprocessing pass.
Many parts are wrapped by helper functions and so the main loop becomes smaller.

remark: selecting "Hide whitespace changes" cuts the diff by ~50%.
image

### CODE REVIEW

File: src/preprocessing/passes/bv_to_int.cpp [hunk 1]
```diff
@@ -271,12 +269,22 @@ Node BVToInt::bvToInt(Node n)
     if (d_bvToIntCache.find(current) == d_bvToIntCache.end())
     {
       // This is the first time we visit this node and it is not in the cache.
+      // We mark this node as visited but not translated by assiging
+      // a null node to it.
       d_bvToIntCache[current] = Node();
+      // all the node's chidlren are added to the stack to be visited
```
Comment (yoni206) [inline src/preprocessing/passes/bv_to_int.cpp:275]:
```suggestion
      // all the node's children are added to the stack to be visited
```

File: src/preprocessing/passes/bv_to_int.cpp [hunk 2]
```diff
@@ -271,12 +269,22 @@ Node BVToInt::bvToInt(Node n)
...
+      // before visigint this node again.
```
Comment (yoni206) [inline src/preprocessing/passes/bv_to_int.cpp:276]:
```suggestion
      // before visiting this node again.
```

File: src/preprocessing/passes/bv_to_int.cpp [hunk 3]
```diff
@@ -288,531 +296,514 @@ Node BVToInt::bvToInt(Node n)
...
+      {
+        throw TypeCheckingException(
+            original.toExpr(),
+            string("Cannot translate to Int: ") + original.toString());
+      }
+      // Now that the translated function symbol was
+      // created, we translate the applicatio and add to the cache.
+      // Additionally, we add
+      // range constraints induced by the original BV width of the
+      // the functions range (codomain)..
```
Comment (yoni206) [inline src/preprocessing/passes/bv_to_int.cpp:673]:
```suggestion

```

File: src/preprocessing/passes/bv_to_int.cpp [hunk 4]
```diff
@@ -288,531 +296,509 @@ Node BVToInt::bvToInt(Node n)
...
+          d_nm->mkNode(kind::MULT, translated_children[0], pow2BvSizeRight);
+      Node b = translated_children[1];
+      returnNode = d_nm->mkNode(kind::PLUS, a, b);
+      break;
+    }
+    case kind::BITVECTOR_EXTRACT:
+    {
+      // ((_ extract i j) a) is a / 2^j mod 2^{i-j+1}
+      // original = a[i:j]
+      Node a = original[0];
```
Comment (ajreynol) [inline src/preprocessing/passes/bv_to_int.cpp:595]: This is not used?

**Review by ajreynol — APPROVED** [review]
LGTM, this is a lot cleaner, one minor comment.

**Pull request merged**